Hide lock token in public webdav responses

PVince81 opened the pull request:
## Description
See https://github.com/owncloud/core/pull/34229#issuecomment-457186641 and https://github.com/owncloud/core/pull/34229#issuecomment-457225278.

## Related Issue
- Fixes https://github.com/owncloud/core/pull/34229#issuecomment-457186641 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?
Manual test, checking that lock discovery never contains a token when queried through public link.
Note that due to https://github.com/sabre-io/dav/pull/1130 not being available yet to OC, instead of having a missing "locktoken" element you'll see one with the value "opaquetoken:" followed by nothing.
This was deemed non-critical in a discussion with @DeepDiver1975 and will solve itself in the next Sabre DAV release + update.

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
<!-- In case of incomplete PR, please list the open tasks here -->
- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

Commit:
Hide lock token in public webdav responses

phil-davis commented:
This changes the behavior of the ``locktoken``` in responses, so we expect some scenario(s) in ``tests/acceptance/features/apiWebdavLocks/publicLink.feature`` to fail.
@individual-it will then adjust the scenarios and push a commit on here (and on the backport).
Then I can review.

davitol reviewed (approved):
👍 Manually tested and works

individual-it commented:
I expected the CI to fail

individual-it reviewed (changes requested):
something is wrong, I expected tests to fail

individual-it commented:
I will adjust the test expectations, currently all tests expect the response to start with/contain `opaquelocktoken:` even the response with the empty token has that

individual-it committed:
adjust acceptance tests

1. check more closely the result to avoid snake-oil tests
2. test PROPFIND on a non-shared folder
